File: BackEnd/src/main/java/com/orange/fintech/group/entity/Group.java
package com.orange.fintech.group.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.orange.fintech.group.dto.GroupCreateDto;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.DynamicInsert;

@Entity
@Getter
@Setter
@DynamicInsert
@Table(name = "trip_group")
// @DynamicUpdate
public class Group {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int groupId;

    @NotNull
    @Column(length = 10)
    private String groupName;

    @NotNull
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd", timezone = "Asia/Seoul")
    private LocalDate startDate;

    @NotNull
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd", timezone = "Asia/Seoul")
    private LocalDate endDate;

    @Column(length = 5)
    private String theme;

    @NotNull
    @ColumnDefault("false")
    private Boolean isCalculateDone;

    public Group(GroupCreateDto dto) {
        this.groupName = dto.getGroupName();
        this.startDate = dto.getStartDate();
        this.endDate = dto.getEndDate();
        this.theme = dto.getTheme();
    }

    public Group() {}

    @Override
    public String toString() {
        return "Group{"
                + "groupId="
                + groupId
                + ", groupName='"
                + groupName
                + '\''
                + ", startDate="
                + startDate
                + ", endDate="
                + endDate
                + ", theme='"
                + theme
                + '\''
                + ", isCalculateDone="
                + isCalculateDone
                + '}';
    }
}

File: FrontEnd/lib/entities/Group.dart
import 'GroupMember.dart';

class Group {
  final int? groupId;
  final String groupName;
  final String theme;
  final String startDate;
  final String endDate;
  bool? groupState;
  final bool isCalculateDone;
  final List<GroupMember> groupMembers;

  Group({
    this.groupId,
    required this.groupName,
    required this.theme,
    required this.startDate,
    required this.endDate,
    required this.groupState,
    required this.isCalculateDone,
    required this.groupMembers,
  });
  factory Group.fromJson(Map<String, dynamic> json) {
    // groupMembers 필드가 JSON에 없기 때문에 빈 리스트로 초기화
    List<GroupMember> membersList = [];

    return Group(
      groupId: json['groupId'],
      groupName: json['groupName'],
      theme: json['theme'] ?? '',  // theme이 null일 경우 기본 값 설정
      startDate: json['startDate'],
      endDate: json['endDate'],
      groupState: json['groupState'],
      isCalculateDone: json['isCalculateDone'],
      groupMembers: membersList,
    );
  }
}